src/database.js

```diff
@@ -54,6 +54,87 @@ function shutdownSQL() {
   }
 }
 
+/**
+ * @async
+ * @function insertNewPackage
+ * @desc Insert a new package inside the DB taking a `Server Object Full` as argument.
+ * @param {object} pack - The `Server Object Full` package.
+ * @returns {object} A Server Status Object.
+ */
+async function insertNewPackage(pack) {
+  try {
+    // TODO: Not sure here if we have to stringify all the package or
+    // only some properties of it.
+    const pack_data = JSON.stringify(pack);
+
+    let command = await sql_storage`
+    INSERT INTO packages (name, creation_method, data)
+    VALUES (${pack.name}, ${pack.creation_method}, ${pack_data})
+    RETURNING pointer;
+    `;
+
+    const pointer = command[0].pointer;
+    if (pointer === undefined) {
+      return {
+        ok: false,
+        content: `Cannot insert ${pack.name} in packages table`,
+        short: "Server Error",
+      };
+    }
+
+    // Populate names table
+    command = await sql_storage`
+    INSERT INTO names
+    (name, pointer) VALUES
+    (${pack.name}, ${pointer});
+    `;
+
+    if (command.count === 0) {
+      return {
+        ok: false,
+        content: `Cannot insert ${pack.name} in names table`,
+        short: "Server Error",
+      };
+    }
+
+    // Populate versions table
+    const latest = pack.releases.latest;
+    const pv = pack.versions;
+
+    // TODO: The following versions handled needs cleanup
+    for (const ver of Object.keys(pv)) {
+      const status = (pv[ver] === latest) ? "latest" : "published";
+
+      const engine = pv[ver].engines;
+      const jsonEngine = JSON.stringify(engine);
+
+      // Save version object into meta, but strip engines and license properties
+      // since we save them in the specific separate columns
+      let meta = pv[ver];
+      delete meta.engines;
+      delete meta.license;
+      const jsonMeta = JSON.stringify(meta);
```

Comment:
```suggestion
      const jsonMeta = meta;
```

Same as before `meta` of `versions` is `JSON_B` so we don't need to stringify.